drivers/sdhc/rcar_mmc.c

```diff
@@ -1569,14 +1569,18 @@ static int rcar_mmc_set_io(const struct device *dev, struct sdhc_io *ios)
 			ret = rcar_mmc_enable_clock(dev, true);
 			break;
 		case SDHC_POWER_OFF:
-			ret = regulator_disable(cfg->regulator_vqmmc);
-			if (ret) {
-				break;
+			if (regulator_is_enabled(cfg->regulator_vqmmc)) {
+				ret = regulator_disable(cfg->regulator_vqmmc);
+				if (ret) {
+					break;
```

Comment:
Probably would be better something like this:
```
if (regulator_disable(cfg->regulator_vqmmc)) {
    break;
}
```

Reply:
it would be good, but it is needed for a returning error code from the function